---
title: Viral and Bacterial Communities of Colorectal Cancer
author:
- name: Geoffrey D Hannigan
  affiliation: 1
- name: Melissa B Duhaime
  affiliation: 2
- name: Mack T Ruffin IV
  affiliation: 3
- name: Charlie C Koumpouras
  affiliation: 1
- name: Patrick D Schloss
  affiliation: 1,\*
address:
  - code: 1
    address: Department of Microbiology & Immunology, University of Michigan, Ann Arbor, Michigan, 48109
  - code: 2
    address: Department of Ecology and Evolutionary Biology, University of Michigan, Ann Arbor, Michigan, 48109
  - code: 3
    address: Department of Family and Community Medicine, Pennsylvania State University Hershey Medical Center, Hershey, Pennsylvania, 17033
  - code: \*
    address: To whom correspondence may be addressed.
output:
  md_document:
    variant: markdown
fontsize: 11pt
mainfont: 'Helvetica'
---

&nbsp;

| ***Corresponding Author Information***
| Patrick D Schloss, PhD
| 1150 W Medical Center Dr. 1526 MSRB I
| Ann Arbor, Michigan 48109
| Phone: (734) 647-5801
| Email: pschloss@umich.edu

| ***Journal***: PNAS ([*Preparation Details*](http://www.pnas.org/site/authors/procedures.xhtml))
| ***Major Classification***: Biological Sciences
| ***Minor Classification***: Microbiology

***Keywords***: Colorectal Cancer, Virome, Machine Learning

\* *Figures included for internal editing purposes*

\newpage

```{r loadfiles, echo=FALSE}
sigfig = 3
alphadf <- read.delim(file = "../rtables/diversity.tsv", header = TRUE, sep = "\t")
twoauc <- read.delim(file = "../rtables/twoclass-auc.tsv", header = TRUE, sep = "\t")
twoss <- read.delim(file = "../rtables/twoclass-auc-statsig.tsv", header = TRUE, sep = "\t")
threeauc <- read.delim(file = "../rtables/threeclass-auc.tsv", header = TRUE, sep = "\t")
threesig <- read.delim(file = "../rtables/threeclass-sig.tsv", header = TRUE, sep = "\t")
idcount <- read.delim(file = "../rtables/idcount.tsv", header = FALSE, sep = "\t")
phagecount <- read.delim(file = "../rtables/phagecount.tsv", header = FALSE, sep = "\t")
corstats <- read.delim(file = "../rtables/cor-stats.tsv", header = TRUE, sep = "\t")
```

\newpage

# Abstract

Colorectal cancer is the second leading cause of cancer-related death in the United States and is a primary cause of morbidity and mortality throughout the world. Although etiology for the majority of colorectal cancer cases remain unclear, its progression has been linked to changes in colonic bacterial community composition. Viruses are another important component of the colonic microbial community. However, they have yet to be studied in colorectal cancer, despite their oncogenic potential. We evaluated the colorectal cancer virome (virus community) using a cohort of 90 human subjects with either healthy, adenomatous (precancerous), or cancerous colons. We utilized 16S rRNA gene, whole shotgun metagenomic, and purified virus metagenomic sequencing methods to compare the colorectal cancer virome to the bacterial community. We found that alpha and beta diversity metrics were insufficient for detecting virome changes in colorectal cancer, but more sophisticated random forest models identified striking changes in the virus community. The majority of the cancer-associated virome consisted of temperate bacteriophages, suggesting that the community was indirectly linked to colorectal cancer by modulating bacterial community structure and function. These results provide foundational evidence that bacteriophage communities are associated with colorectal cancer and likely impact cancer progression by altering the bacterial host communities.

**Word Count**: 201 / 250

# Significance Statement
Colorectal cancer is a leading cause of cancer-related death in the United States and worldwide. Its progression and severity have been linked to colonic bacterial community composition. Although viruses have been linked to other cancers and diseases, little is known about colorectal cancer virus communities. We addressed this knowledge gap by identifying changes in colonic virus communities in colorectal cancer patients and how they compared to bacterial community changes. The results suggested an indirect role for the virome impacting colorectal cancer by modulating their associated bacterial community. These findings both support a biological role for viruses in colorectal cancer, as well as provide a new understanding of basic colorectal cancer etiology.

**Word Count**: 111 / 120

\newpage

# Introduction
Due to their mutagenic abilities and propensity for functional manipulation, human viruses are strongly associated with, and in many cases cause, cancer [@Feng:2008kr; @Shuda:2011gf; @Schiller:2012ba; @Chang:1994up]. Because bacteriophages are crucial for bacterial community stability and composition [@Harcombe:2005fd; @RodriguezValera:2009cr; @Cortez:2014bk], and because bacteria have been implicated as oncogenic agents [@Zackular:2014fba; @Garrett:2015fg; @Baxter:2014hb], bacteriophages have the potential to indirectly impact cancer. The gut virome (the virus community of the gut) therefore has the potential to impact health and disease (e.g. cancer).[Comment: the gut virome isn't associted with the following diseases...] The human virome has already been associated with diseases including periodontal disease [@Ly:2014ew], HIV [@Monaco:2016ita], antibiotic exposure [@Abeles:2015dy; @Modi:2013fi], urinary tract infections [@SantiagoRodriguez:2015gd], and inflammatory bowel disease [@Norman:2015kb]. The precedence of the virome impacting human health and the strong association of bacterial communities with colorectal cancer suggests that colorectal cancer may be associated with altered virus communities.

Colorectal cancer is the second leading cause of cancer-related deaths in the United States [@Siegel:2014jo]. The US National Cancer Institute estimates over 1.5 million Americans have been diagnosed with colorectal cancer in 2016, and over 500,000 Americans died from the disease [@Siegel:2014jo]. Although the etiology for [cite specific number - 80%?] of colorectal cancer cases remains unclear, variation in colorectal bacterial communities have been linked to the disease [@Flynn:2016iu; @Zackular:2016en; @Zackular:2014fba; @Baxter:2014hb]. This work has led to a proposed disease model in which bacteria colonize the colon, develop biofilms, promote inflammation, and enter an oncogenic synergy with the cancerous human cells [@Flynn:2016iu]. This association has also allowed researchers to leverage bacterial community signatures as biomarkers to provide accurate, noninvasive colorectal cancer detection from stool [@Zackular:2014fba; @Baxter:2016dja]. While an understanding of colorectal cancer bacterial communities has proven fruitful both for disease classification and understanding underlying etiology, bacteria are only a subset of the colon microbiome. Viruses are another important component of the colon microbial community that have yet to be studied in the context of colorectal cancer. We evaluated disruptions in virus and bacterial community composition in a human cohort whose stool was sampled at the three relevant stages of cancer development: healthy, adenomatous, and cancerous.

Colorectal cancer is a stepwise process that begins when healthy tissue develops into a precancerous polyp (i.e. adenoma) in the large intestine [@Fearon:2011go]. If left untreated, the adenoma will develop into a cancerous lesion that can invade and metastasize, leading to severe illness and death. Progression to cancer can be prevented when precancerous adenomas are detected and removed during routine screening [@Levin:2008kz; @Zauber:2015cr], and survival for colorectal cancer patients may exceed 90% when the lesions are detected early and removed [@Levin:2008kz]. Thus work that aims to facilitate early detection and prevention of progression beyond early cancer stages has great potential to inform therapeutic development.

Here we address the gap in knowledge of whether virus communities are altered in colorectal cancer and, if they are, how those changes might impact cancer progression and severity. We also aimed to evaluate their potential as a reservoir of diagnostic biomarkers. We report that colorectal cancer is associated with variation in colonic virus community composition. The majority of viruses identified within the virome were temperate bacteriophages. Just as the association between the bacterial community and colorectal cancer was driven by select influential bacteria including *Fusobacterium*, the association between the virome and colorectal cancer was driven by a subset of influential phages. Our data suggest that the influential phages do not exclusively infect influential bacterial, but rather act through the community as a whole. The implications of these findings are threefold. *First*, this supports a biological role for the virome in colorectal cancer development and suggests that more than bacteria are involved in the process. *Second*, we present a supplementary, or even alternative virus-based approach for classification modeling of colorectal cancer using stool samples. *Third*, we provide initial support for the importance of studying the virome as a component of the microbiome ecological network, especially in cancer. [Comment: I worry that from "We report..." to the end of the paragraph is too much a rehash of the Results. I would suggest moving this section to the abstract since the abstract is a bit light on results. ]

# Results

## Cohort Design, Sample Collection, and Processing
The study cohort consisted of 90 human subjects, 30 of which had healthy colons, 30 of which had adenomas, and 30 which had carcinomas **(Figure \ref{sampleproc})**. Half of the stool was used to sequence the bacterial communities using both 16S rRNA gene and shotgun sequencing techniques. The other half of the stool samples were purified for virus like particles (VLPs) before genomic DNA extraction and shotgun metagenomic sequencing. The VLP purification allowed us to observe the *active virome* because we only sequenced those viruses that were encapsulated.

Virus DNA was purified prior to sequencing, allowing us to analyze DNA exclusively from within virus capsids **(Figure \ref{sampleproc})**. Each extraction protocol was performed with a blank control to detect contaminants from reagents or other unintentional sources. Only one of the nine controls contained DNA at a minimal concentration (XX ng/ul), thus providing initial evidence of successful sequencing of VLP genomic DNA over potential contaminants **(Figure \ref{qualcontrol} A)**. As was expected, these controls yielded few sequences and were mostly removed while rarefying the datasets to a common number of sequences **(Figure \ref{qualcontrol} B)**. The high quality phage and bacterial sequences were assembled into highly covered contigs longer than 1kb **(Figure \ref{contigqc})**. Because contigs only represent genome fragments, we further clustered related contigs into operational genomic units (OGUs) **(Figure \ref{contigqc} - \ref{clustercontigqc})**.
[Comment: What do you think about using operational viral units (OVU) rather than OGU? I think of OGU as representing bacterial genomes. Adding phage to the name would make it more clear what we are refering to]

## Unaltered Virome Diversity in Colorectal Cancer
Microbiome and disease associations are often described as being of an altered diversity (i.e. "dysbiotic"). We therefore initially evaluated the changes in virome OGU diversity in colorectal cancer. We evaluated differences in communities between disease states using the Bray-Curtis, Shannon entropy, and richness diversity metrics. ~~To control for uneven sequencing depths, we rarefied to a minimum depth that maintained most samples while excluding the sparsely sequenced blank controls.~~ [Comment: put this in the methods section rarefied != sub-sampled, make sure you rarefied (i.e. subsampled 1000 times)] We observed no significant alterations in either Shannon entropy or richness in the diseased state **(Figure \ref{betaogu} C-D)**. There was also no statistically significant clustering of the disease groups (ANOSIM p-value = `r signif(alphadf[c(alphadf$Title %in% "anovawithoutneg"),"Stat"], digits = sigfig)`, **Figure \ref{betaogu}**). It is worth noting that there was a significant difference between the few blank controls that remained after rarefying the data, and the other study groups (ANOSIM p-value = `r signif(alphadf[c(alphadf$Title %in% "anovawithneg"),"Stat"], digits = sigfig)`, **Figure \ref{betaogunegative})**. This further supported the quality of our sample set. Similar to what was previously seen when analyzing the 16S rRNA gene sequences from the same samples [@Zackular:2014fba], standard alpha and beta diversity metrics were insufficient for capturing virus community differences between disease states.

## Altered Virome Composition in Colorectal Cancer
16S rRNA gene relative abundance profiles are effective feature sets for classifying stool samples as originating from individuals with healthy, adenomatous, or cancerous colons [@Zackular:2014fba; @Baxter:2016dja]. The exceptional performance of bacteria in these classification models supports a role for bacteria in colorectal cancer. We built off of these findings by evaluating the ability of virus community signatures to classify stool samples and compared their performance to models built using bacterial community signatures.

To identify the altered virus communities associated with colorectal cancer, we built and tested random forest models for classifying stool samples as belonging to individuals with either cancerous or healthy colons. We confirmed that our bacterial 16S rRNA gene model replicated the performance of the original report which used logit models instead of random forest models **(Figure \ref{predmodel} A)** [@Zackular:2014fba]. We then compared the bacterial 16S rRNA gene model to a model built using virome relative abundance. Statistically, the viral model performed better than the bacterial model (corrected p-value = `r signif(twoss[c(twoss$First %in% "Virus" & twoss$Second %in% "Bacteria"),"pval"], digits = sigfig)`), although the results were practically similar, with the viral and bacterial models achieving mean AUC (area under the curve) values of `r signif(twoauc[c(twoauc$class %in% "Virus"),"meanAUC"], digits = sigfig)` and `r signif(twoauc[c(twoauc$class %in% "Bacteria"),"meanAUC"], digits = sigfig)`, respectively **(Figure \ref{predmodel} A - B)**.

[Comment:
1A: Get rid of the legend title "Disease". I feel like the 0.1 and 0.9 marks on the ROC curves adds clutter without improving interpretation.
1B: Can you order: Bacteria, Virus, Combined, Metagenome - why is the combined AUC > virus when it seems the opposite in A? Can you put the AUCs in the legend for 1A and get rid of 1B?
1C: Format OTU names to be like "OTU 4" rather than "Otu00004"
1D: Format cluster  names to be like "Cluster 115" rather than "Cluster_115"
1E: See comments for 1C and 1D
Can you justify the y-axis in C, D, and E so they are vertically aligned and the x-axes are the same lengths?
]

To evaluate the ability of both bacterial and viral biomarkers to classify samples, we built a combined model that used both bacterial and viral community data. The combined model yielded statistically significant but small performance improvement beyond the viral (corrected p-value = `r signif(twoss[c(twoss$First %in% "Virus" & twoss$Second %in% "Combined"),"pval"], digits = sigfig)`) and bacterial (corrected p-value = `r signif(twoss[c(twoss$First %in% "Combined" & twoss$Second %in% "Bacteria"),"pval"], digits = sigfig)`) models, yielding an AUC of `r signif(twoauc[c(twoauc$class %in% "Combined"),"meanAUC"], digits = sigfig)` **(Figure \ref{predmodel} A - B)**. This suggested that features from the virus and bacterial communities interact to improve the ability to classify stool as belonging to individuals with cancerous colons.

To determine the advantage of viral metagenomic methods over bacterial metagenomic methods, we compared the viral model to a model built using relative abundance profiles from bacterial metagenomic shotgun sequencing data. This model performed more poorly than any of the other models (mean AUC = `r signif(twoauc[c(twoauc$class %in% "Metagenomic"),"meanAUC"], digits = sigfig)`) **(Figure \ref{predmodel} A - B)**. Further investigation revealed that the bacterial 16S rRNA gene model was strongly driven by sparse and low abundance OTUs **(Figure \ref{16scompare})**. Removal of OTUs with a median abundance of zero resulted in the removal of six OTUs, and a loss of model performance down to what was observed in the metagenome-based model **(Figure \ref{16scompare} A)**. The majority of these OTUs had a relative abundance lower than 1% **(Figure \ref{16scompare} B)**. [Comment: Can you add a comment about the nature of the features in the viral model?]

The association between the two communities and colorectal cancer was driven by a few important microbes, measured using the mean decrease in model accuracy when each was removed. *Fusobacterium* was the primary driver of the bacterial association with colorectal cancer, which was consistent with its previously described oncogenic potential **(Figure \ref{predmodel} C)**[@Flynn:2016iu]. The virome signature was also driven by a few OGU, suggesting a role for the OGUs in cancer development **(Figure \ref{predmodel} D)**. The identified viruses were bacteriophages, belonging to *Siphoviridae*, *Myoviridae*, and orphan phage taxa without taxonomic identifiers (denoted "unclassified"). Many of the important viruses were unidentifiable (denoted "unknown"), suggesting they are members of the abundant unknown viral population associated with the human virome. This is common in the virome; studies can have as much as 95% of virus sequences belong to unknown genomic units [@Pedulla:2003tu; @Hannigan:2015fz]. When the bacterial and viral community signatures were combined, both bacterial and viral organisms drove the community association with cancer **(Figure \ref{predmodel} E)**.

## Shifted Phage Influence Across Cancer Stages
Because our cohort included people with healthy, adenomatous, and cancerous colons we were able to evaluate shifts in influential phage identities during cancer progression. We evaluated community shifts between the two disease stage transitions (healthy to adenomatous and adenomatous to cancerous) by building random forest models to compare only the diagnosis groups around the transitions. While bacterial 16S rRNA gene models performed equally well for all disease class comparisons, the virome model performances differed **(Figure \ref{transitionmodels} A-B)**. Like bacteria **(Figure \ref{transitionmodels} F-H)**, different virome members were important in the transitions from healthy to adenomatous and adenomatous to cancerous stages **(Figure \ref{transitionmodels} C-E)**.

After evaluating our ability to classify samples between two disease states, we performed a three-class random forest model including all disease states. The 16S rRNA gene model model yielded a mean AUC of `r signif(threeauc[c(threeauc$class %in% "Bacteria"),"meanauc"], digits = sigfig)` and outperformed the viral community model which yielded a mean AUC of `r signif(threeauc[c(threeauc$class %in% "Virus"),"meanauc"], digits = sigfig)` (p-value = `r signif(threesig, digits = sigfig)`, **Figure \ref{threewaymodel} A-C**). The microbes important for the cancer vs healthy and healthy vs adenoma models were also important for the three-class model **(Figure \ref{threewaymodel} D-E)**. The most important bacterium in the two and three class models was the same *Fusobacterium* (OTU 4) **(Figure \ref{predmodel} C, Figure \ref{threewaymodel} D)**. The viruses most important to the three-class model were also identified as bacteriophages **(Figure \ref{predmodel} D, Figure \ref{threewaymodel} E)**. {Influential phage relative abundances were both higher and lower in cancer **(Figure \ref{threewaymodel} F)**. Not all important OGUs were of increased abundance in the diseased state. The viral classification model depended on the unique signatures of these different abundance profiles to accurately classify each sample.} [Comment: I don't feel like the text between the {}'s says anything. It's like word salad.]

[Comment: Can't distinguish colors in Fig S9F (and axis label overlaps panel label)]

## Bacteriophage Dominance in Colorectal Cancer Virome
Changes in the colorectal cancer virome could have been driven directly by eukaryotic viruses or indirectly by bacteriophages acting through their bacterial hosts. To better understand the types of viruses that are important for colorectal cancer, we identified the virome OGUs as being similar to either eukaryotic viruses or bacteriophages. The most important viruses to the classification model were identified as bacteriophages (**Figure \ref{threewaymodel})**. Overall we were able to identify `r signif(idcount, digits = sigfig)`% of the OGUs as known viruses, and `r signif(phagecount, digits = sigfig)`% of those viral OGUs aligned to bacteriophage reference genomes.

We evaluated whether the phages in the community were primarily lytic (replicate by lysing their hosts) or temperate (lysogenic; able to integrate into their host's genome, as well as lyse the cell). We accomplished this by identifying three markers for temperate phages in the OGU representative sequences: 1) presence of phage integrase genes, 2) presence of known prophage genes, according the the ACLAME (A CLAssification of Mobile genetic Elements) database, and 3) nucleotide similarity to regions of bacterial genomes, as previously described [@Minot:2011ez; @Hannigan:2015fz]. We found that the majority of the colon phages were temperate, and that the overall fraction of temperate phages remained consistent throughout the healthy, adenomatous, and cancerous stages **(Figure \ref{replicationstyles} E)**. Thus the majority of the OGUs are temperate bacteriophages and not eukaryotic viruses indicating that the association between the virome and colorectal cancer is reliant on bacteriophage communities that can lie dormant in bacterial genomes. These findings were consistent with previous reports suggesting the gut virome is primarily composed of temperate phages [@Minot:2011ez; @Monaco:2016ita; @Norman:2015kb; @Reyes:2010cwa].

## Community Context of Influential Phages
Because the link between colorectal cancer and the virome was driven by bacteriophages, we hypothesized that the influential phages were primarily predators of the influential bacteria, and thus influenced their relative abundance through predation. If this hypothesis were true, we would expect a correlation between the relative abundances of influential bacteria and phages. Instead we observed a strikingly low correlation between the bacterial and phage relative abundances **(Figure \ref{correlations} A,C)**. There was an overall absence of correlation between the most influential phage OGUs and bacterial OTUs **(Figure \ref{correlations} B)**. This evidence supported our null hypothesis that the influential phages were not primarily predators of influential bacteria.

[ Comment: how are the rows and columns sorted in Figure 2? Looks like by importance, how about sorting them by co-occurrence? ]

Given these findings, we hypothesized that the most influential phages were acting by infecting a wide range of bacteria in the overall community, instead of just the influential bacteria. This hypothesis would be supported by showing that the influential bacteriophages were community hubs within the bacteria and phage interactive network. We investigated the infectious capabilities and potential host ranges of all phage OGUs using a random forest model to predict which phages infected which bacteria in the overall community (**Cite Network Preprint**). The predicted interactions were then used to identify community hub phages. This revealed a wide tropism range for the bacteriophages within the community **(Figure \ref{network} A)**. We calculated the alpha centrality (measure of importance in the ecosystem network) of each phage OGU's connection to the rest of the network, and compared the centrality to the importance of each OGU in the colorectal cancer classification model. The phages with high centrality values were defined as community hubs. Phage OGU centrality was significantly and positively correlated with importance to the disease model (p-value = `r signif(corstats[c(corstats$names %in% "pval"),"scores"], digits = sigfig)`, R = `r signif(corstats[c(corstats$names %in% "rho"),"scores"], digits = sigfig)`), suggesting that phages important in driving colorectal cancer were also more likely to be community hubs **(Figure \ref{network} B)**. Together these findings supported our hypothesis that influential phages were hubs within their microbial communities.

# Discussion
Because of their propensity for mutagenesis and capacity for modulating their host functionality, many viruses are oncogenic [@Feng:2008kr; @Shuda:2011gf; @Schiller:2012ba; @Chang:1994up]. Some bacteria also have oncogenic properties, meaning bacteriophages may play an indirect role in promoting carcinogenesis by influencing bacterial communities [@Zackular:2014fba; @Garrett:2015fg; @Baxter:2014hb]. Despite their carcinogenic potential and the strong association between bacteria and colorectal cancer, the link between virus colorectal communities and colorectal cancer has yet to be evaluated. Here we show that, like colonic bacterial communities, the colon virome was altered in colorectal cancer. Our findings support a working hypothesis for oncogenesis by phage-modulated bacterial community composition.

Our data allowed us to begin delineating the role the colonic virome played in colorectal cancer **(Figure \ref{modelsummary} A)**. We found basic diversity metrics of alpha diversity (richness and Shannon entropy) and beta diversity (Bray-Curtis dissimilarity) were insufficient for identifying virome community changes between healthy and cancerous states. By implementing a more sophisticated machine learning approach (random forest classification), we detected strong associations between the colon virus community composition and colorectal cancer. The colorectal cancer virome was composed primarily of bacteriophages. These phage communities were not exclusive predators of the most influential bacteria, as demonstrated by the lack of correlation between the abundances of the bacterial and phage populations. Instead, we identified influential phages as being community hubs, suggesting phages influence cancer by altering the greater bacterial community instead of directly modulating the influential bacteria. Our previous work has shown that modifying colon bacterial communities alters colorectal cancer progression and tumor burden in mice [@Zackular:2016en; @Baxter(the Microbiome article)]. This provides a precedent for phage indirectly influencing colorectal cancer progression by altering the bacterial community composition. Overall, our data supports a model in which the bacteriophage community modulates the bacterial community, and through those interactions indirectly influences the bacteria driving colorectal cancer progression **(Figure \ref{modelsummary} A)**. Although our evidence suggested phages indirectly influenced colorectal cancer development, we were not able to rule out the role of phages directly interacting with the human host.

In addition to modeling the basic potential connections between virus communities, bacteria communities, and colorectal cancer, we also used our data and existing knowledge of phage biology to develop a working hypothesis for the mechanisms by which this may occur. This was done by incorporating our findings into the current model for colorectal cancer development **(Figure \ref{modelsummary} B)** [@Flynn:2016iu]. We hypothesize that the broadly infectious phages in the colon began lysing, and thereby disrupting, the bacterial communities to open a niche in which opportunistic bacteria (such as *Fusobacterium nucleatum*) were able to colonize. Once the initial influential bacteria had established themselves in the epithelium, other opportunistic bacteria were able to adhere to the driver, colonize, and begin establishing a biofilm. Phages may have played a role in biofilm dispersal and growth by lysing bacteria within the biofilm, a process important for effective biofilm growth [@Rossmann:2015cj]. The oncogenic bacteria were then able to transform the epithelial cells and disrupt tight junctions to infiltrate the epithelium, thereby initiating an inflammatory immune response. As the adenomatous polyps developed and progressed towards carcinogenesis, we observed a shift in the phages and bacteria whose relative abundance was most influential. As the bacteria entered their oncogenic synergy with the epithelium, we hypothesize the phages could have continued mediating biofilm dispersal, as well as support the colonized oncogenic bacteria by lysing competing cells to maintain the niche and provide nutrients to the other bacteria. In addition to highlighting the most likely mechanisms by which the colorectal cancer virome is interacting with the bacterial communities, this outline will guide future research investigations of the role of the virome in colorectal cancer.

A notable observation from our analysis was the poor performance observed using bacterial metagenomic methods compared to the performance of models using viral metagenomes or 16S rRNA gene sequences. This observation highlights the importance of high sequencing coverage in bacterial metagenomic studies, and the advantage of 16S rRNA gene sequencing over whole metagenomic shotgun sequencing. We found that there were six bacterial OTUs that drove the performance of the 16S rRNA gene classification model, and these OTUs were all sparsely present and lowly abundant. Filtration of OTUs with a median relative abundance of zero resulted in the removal of the six important OTUs and reduced model performance to being nearly random like the bacterial metagenomic model. The bacterial metagenomic OGUs represented only the most abundant taxa, which was not informative for this application. There has been some success in using shotgun metagenomic approaches for stool colorectal cancer classification, but these approaches did not utilize OGU clustering like we did here, and the models only performed **as well** as the 16S rRNA gene model [@Zeller:2014ix]. Thus the targeted 16S rRNA gene sequencing approach, which yielded only a fraction of the bacterial metagenomic sequences, was more effective for detecting colorectal cancer in stool samples. Despite a loss of enthusiasm for 16S rRNA gene sequencing in favor of shotgun metagenomic techniques, 16S rRNA gene sequencing is still a superior methodological approach for some important applications.

In addition to the therapeutic ramifications for understanding the colorectal cancer microbiome, our findings provide a proof-of-principle that viruses, while under-appreciated and understudied in the human microbiome, are an important contributor to human disease that has the potential to provide an abundance of information that supplements that of bacterial communities. Evidence has suggested that the virome is a crucial component to the microbiome and that bacteriophages are important players. Bacteriophage and bacterial communities cannot thrive without each other [@RodriguezValera:2009cr]. Not only is the human virome an important part of human health and disease, but it appears to have a particular significance in cancer research.

# Methods

## Analysis Source Code & Availability
All associated source code is available at the following GitHub repository: https://github.com/SchlossLab/Hannigan-2016-ColonCancerVirome.

## Study Design and Patient Sampling
This study was approved by the University of Michigan Institutional Review Board and all subjects provided informed consent. Design and sampling of this sample set have been reported previously [@Zackular:2014fba]. Briefly, whole evacuated stool was collected from patients who were 18 years of age or older, able to provide informed consent, have had colonoscopy and histologically confirmed colonic disease status, had not had surgery, had not had chemotherapy or radiation, and were free of known co-morbidities including HIV, chronic viral hepatitis, HNPCC, FAP, and inflammatory bowel disease. Samples were collected from four locations: Toronto (Ontario, Canada), Boston (Massachusetts, USA), Houston (Texas, USA), and Ann Arbor (Michigan, USA). Ninety patients were recruited to the study, thirty of which were designated healthy, thirty with detected adenomas, and thirty with detected carcinomas.

## 16S rRNA Gene Sequence Data Acquisition & Processing
The 16S rRNA gene sequences associated with this study were previously reported [@Zackular:2014fba]. Sequence (fastq) and metadata files were downloaded from http://www.mothur.org/MicrobiomeBiomarkerCRC. The 16S rRNA gene sequences were analyzed as described previously, relying on the mothur software package (v1.37.0) [@Kozich:XXXXXX; @Schloss:2009do]. Briefly, the sequences were de-replicated, aligned to the SILVA database [@Pruesse:2007jc], screened for chimeras using UCHIME [@Edgar:2011gy], and binned into operational taxonomic units (OTUs) using a 97% similarity threshold. Abundances were normalized for uneven sequencing depth by randomly sub-sampling to 10,000 sequences, as previously reported [@Baxter:2016dja].

## Whole Metagenomic Library Preparation & Sequencing
DNA was extracted from stool samples using the PowerSoil-htp 96 Well Soil DNA Isolation Kit (Mo Bio Laboratories) using an EPMotion 5075 pipetting system. Purified DNA was used to prepare a shotgun sequencing library using the Illumina Nextera XT library preparation kit according to the standard kit protocol. The tagmentation time was increased from five minutes to ten minutes to improve DNA fragment length distribution. The library was sequenced using one lane of the Illumina HiSeq4000 platform and yielded 125 bp paired end reads.

## Virus Metagenomic Library Preparation & Sequencing
Genomic DNA was extracted from purified virus-like particles (VLPs) from stool samples, using a modified version of a previously published protocol [@Hannigan:2015fz]. Briefly, an aliquot of stool (ca. 0.1g) was resuspended in SM [Comment: Define SM] buffer and vortexed to facilitate resuspension. The resuspended stool was centrifuged to remove major particulate debris, followed by filtering through a 0.22-Âµm filter to remove smaller contaminants. The filtered supernatant was treated with chloroform to lyse contaminating cells including bacteria, human, fungi, etc. The exposed genomic DNA from the lysed cells was degraded by treating the samples with DNase. The DNA was extracted from the purified VLPs using the Wizard PCR Purification Preparation Kit (Promega). Disease classes were staggered across purification runs to prevent run variation as a confounding factor. Purified DNA was used to prepare a shotgun sequencing library using the Illumina Nextera XT library preparation kit according to the standard kit protocol. The tagmentation time was increased from five minutes to ten minutes to improve DNA fragment length distribution. The PCR cycle number was increased from twelve to eighteen cycles to address the low biomass of the samples, as has been described previously [@Hannigan:2015fz]. The library was sequenced using one lane of the Illumina HiSeq4000 platform and yielded 125 bp paired end reads.

## Metagenome Quality Control
Both the viral and whole metagenomic sample sets were subjected to the same quality control procedures. The sequences were obtained as de-multiplexed fastq files from the HiSeq platform and subjected to 5' and 3' adapter trimming using the CutAdapt program (v1.9.1) with an error rate of 0.1 and an overlap of 10 [@Martin:2011eu]. The FastX toolkit (v0.0.14) was used to quality trim the reads to a minimum length of 75bp and a minimum quality score of 30 [@FASTXToolkit:wr]. Reads mapping to the human genome were removed using the DeconSeq algorithm (v0.4.3) and default parameters [@Schmieder:2011fo].

## Contig Assembly & Abundance
Contigs were assembled using paired end read files that were purged of sequences without a corresponding pair (e.g. One read removed due to low quality). The Megahit program (v1.0.6) was used to assemble contigs for each sample using a minimum contig length of 1000 bp and iterating assemblies from 21-mers to 101-mers by 20 [@Li:2016kd]. Contigs from the virus and whole metagenomic sample sets were concatenated within their respective groups. Abundance of the contigs within each sample was calculated by aligning sequences back to the concatenated contig files using the bowtie2 global aligner (v2.2.1), with a 25 bp seed length and an allowance of one mismatch [@Langmead:2012jh]. Abundance was corrected for contig reference length and the number of contigs included in each operational genomic unit. Abundance was also corrected for uneven sampling depth by randomly sub-sampling virome and whole metagenomes to 1,000,000 and 500,000 reads, respectively, and removing samples with less total samples than the threshold. Thresholds were set for maximizing sequence information while minimizing numbers of lost samples.

## Operational Genomic Unit Classification
Much like operational taxonomic units (OGUs) are used as an operational definition of similar 16S rRNA gene sequences in absence of taxonomic identification, we operationally defined closely related contig sequences as operational genomic units (OGUs) in the absence of taxonomic identity. OGUs were defined with the CONCOCT algorithm (v0.4.0) which bins related contigs by similar tetra-mer and co-abundance profiles within samples using a variational Bayesian approach [@Alneberg:2014fc]. CONCOCT was used with a length threshold of 1000 bp for virus contigs and 2000 bp for bacteria due to computational limitations.

## Diversity
Alpha and beta diversity were calculated using the operational genomic unit abundance profiles for each sample. Sequences were rarefied to 100,000 sequences. Samples with less than the cutoff were removed from the analysis. Alpha diversity was calculated using the Shannon Entropy and Richness metrics. Beta diversity was calculated using the Bray-Curtis metric (mean of 25 random sub-sampling iterations), and the statistical significance between the disease state clusters was assessed using an analysis of similarity (ANOSIM) with a post-hoc multivariate Tukey test. All diversity calculations were performed in R using the Vegan package [@veganCommunityEco:xdLliqSE.

## Classification Modeling
Classification modeling was performed in R using the Caret package [@caretClassificatio:_U2Lit_1]. OTU and OGU abundance data was preprocessed by removing features (OTUs and OGUs) that were present in less than half of the samples. This served both as an effective feature reduction technique and made the calculations computationally feasible. The binary random forest model was trained using the Area Under the ROC Curve (AUC) and the three-class random forest model was trained using the mean AUC. Both were validated using five-fold cross validation. Each training set was repeated five times, and the model was tuned across five iterations of mtry values. For consistency and accurate comparison between feature groups (e.g. bacteria, virus), the sample model parameters were used for each group. The maximum AUC during training was recorded across 10 iterations of each group model creation to test the significance of the differences between feature set performance. Statistical significance was evaluated using a Wilcoxon test between two categories, or a pairwise Wilcoxon test with Bonferroni corrected p-values when comparing more than two categories.

## Taxonomic Identification of Operational Genomic Units
Viral operational genomic units (OGUs) were identified using a reference database consisting of all bacteriophage and eukaryotic virus genomes present in the European Nucleotide Archives. The longest contiguous sequence in each operational genomic unit was used as a representative sequence for classification. Each representative sequence was aligned to the reference genome database using the tblastx alignment algorithm (v2.2.27) and a strict similarity threshold (e-value < 1e-25) [@Camacho:2009fc]. Annotation was interpreted as phage, eukaryotic virus, or unknown.

## Ecological Network Analysis & Correlations
The ecological network of the bacterial and phage operational genomic units were constructed and analyzed as previously described (cite network preprint here). Briefly, a random forest model was used to predict interactions between bacterial and phage genomic units, and those interactions were recorded in a graph database using *neo4j* graph databasing software (v2.3.1). The degree of phage centrality was quantified using the alpha centrality metric in the igraph CRAN package. A Spearman correlation was performed between model importance and phage centrality scores.

## Phage Replication Style Identification
Phage OGU replication sytle was identified using methods described previously [@Minot:2011ez; @Hannigan:2015fz; @Hannigan:2017ky]. Briefly, we identified lysogenic phage OGUs as representative contigs containing at least one of three genomic markers: 1) phage integrase genes, 2) prophage genes from the ACLAME database, 3) genomic similarity to bacterial reference genomes. Integrase genes were identified in phage OGU representative contigs by aligning the contigs to a reference database of all known phage integrase genes from the Uniprot database (Uniprot search term: "organism:phage gene:int NOT putative"). Prophage genes were identified in the same way, using the ACLAME set of reference prophage genes. In both cases, the blastx algorithm was used with an e-value of 10e-5. Representative contigs were also identified as potential lysogenic phages by having a high genomic similarity to bacterial genomes. To accomplish this, representative phage contigs were aligned to the European Nucleotide Archive bacterial genome reference set using the blastn algorithm (e-value < 10e-25).

# Conflicts of Interest
The authors declare no conflicts of interest.

# Acknowledgments
The authors thank the Great Lakes-New England Early Detection Research Network for providing the fecal samples that were used in this study. GD Hannigan was supported by the Michigan Molecular Mechanisms of Microbial Pathogenesis Fellowship. PD Schloss was supported by funding from the National Institutes of Health (P30DK034933). MT Ruffin was supported by funding from the National Institutes of Health (5U01CA86400).

\newpage

# Figures

![*Results from healthy vs cancer classification models built using virome signatures, bacterial 16S rRNA gene sequence signatures, whole metagenomic signatures, and a combination of virome and 16S rRNA gene sequence signatures. A) ROC curve for visualizing the performance of each of the models for classifying stool as coming from either an individual with a cancerous or healthy colon. B) Quantification of the AUC variation for each model, and how it compared to each of the other models based on 15 iterations. A pairwise Wilcoxon test with a false discovery rate multiple hypothesis correction demonstrated that all models are significantly different from each other (p-value < 0.01). C) Mean decrease in accuracy (measurement of importance) of each operational taxonomic unit within the 16S rRNA gene classification model when removed from the classification model. Results based on 15 iterations. Mean is represented by a point, and bars represent standard error. D) Mean decrease in accuracy of each operational genomic unit in the virome classification model. E) Mean decrease in accuracy of each operational genomic unit and operational taxonomic unit in the model using both 16S rRNA gene and virome features.*\label{predmodel}](../figures/predmodel-viromebacteria.pdf){ width=100% }

\newpage

![*Relative abundance correlations between bacterial OTUs and virome OGUs. A) Pearson correlation coefficient values between all bacterial OTUs (x-axis) and viral OGUs (y-axis) with blue being positively correlated and red being negatively correlated. Operational units are organized by importance in their colorectal cancer classification models, such that the most important units are in the top left corner. B) Magnification of the boxed region in pannel (A), highlighting the correlation between the most important bacterial OTUs and virome OGUs. The most important operational units are in the top left corner of the heatmap, and the correlation scale is the same as pannel (A). C) Histogram quantifying the frequencies of Pearson correlation coefficients between all bacterial OTUs and virome OGUs.*\label{correlations}](../figures/phage-bacteria-cor.pdf){ width=100% }

\newpage

![*Community network analysis utilizing predicted interactions between bacteria and phage operational genomic units. A) Visualization of the community network for our colorectal cancer cohort. B) Scatter plot illustrating the correlation between importance (mean decrease in accuracy) and the degree of centrality for each OGU. A linear regression line was fit to illustrate the correlation (blue) which was found to be statistically significantly and weakly correlated (p-value = `r signif(corstats[c(corstats$names %in% "pval"),"scores"], digits = sigfig)`, R = `r signif(corstats[c(corstats$names %in% "rho"),"scores"], digits = sigfig)`).*\label{network}](../figures/NetworkAndScatter.pdf){ width=80% }

\newpage

![*Final model and working hypothesis from this study. A) Basic model illustrating the connections between the virome, bacterial communities, and colorectal cancer. B) Working hypothesis of how the bacteriophage community is associated with colorectal cancer and the associated bacterial community.*\label{modelsummary}](../figures/WorkingHypothesis.pdf){ width=100% }

\newpage

# Supplemental Figures

\beginsupplement

![*Cohort and sample processing outline. Thirty subject stool samples were collected from healthy, adenoma (pre-cancer), and carcinoma (cancer) patients. Stool samples were split into two aliquots, the first of which was used for bacterial sequencing and the second which was used for virus sequencing. Bacterial sequencing was done using both 16S rRNA amplicon and whole metagenomic shotgun sequencing techniques. Virus samples were purified for viruses using filtration and a combination of chloroform (bacterial lysis) and DNase (exposed genomic DNA degradation). The resulting encapsulated virus DNA was sequenced using whole metagenomic shotgun sequencing.* \label{sampleproc}](../figures/sample-processing.pdf){ width=100% }

\newpage

![*Basic Quality Control Metrics. A) VLP genomic DNA yield from all sequenced samples. Each bar represents a sample which is grouped and colored by its associated disease group. B) Sequence yield following quality control including quality score filtering and human decontamination.*\label{qualcontrol}](../figures/qualitycontrol.pdf){ width=100% }

\newpage

![*Length and coverage statistics. A) Heated scatter plot demonstrating the distribution of contig coverage (number of sequences mapping to each contig) and contig length for the virus metagenomic sample set. B) Scatter plot illustrating the distribution of operational genomic unit (OGU) length and sequence coverage for the virus metagenomic sample set. C) Heated scatter plot demonstrating the distribution of contig coverage and length for the whole metagenomic sample set. D) Scatter plot illustrating the distribution of operational genomic unit (OGU) length and sequence coverage for the whole metagenomic sample set.*\label{contigqc}](../figures/qc-contiglengthcoverage.pdf){ width=75% }

\newpage

![*Operational genomic unit composition stats. A) Strip chart demonstrating the length and frequency of contigs within each operational genomic unit of the virome sample set. The y-axis is the operational genomic unit identifier, and x-axis is the length of each contig, and each dot represents a contig found within the specified operational genomic unit. B) Density plot (analogous to histogram) of the number of virome operational genomic units containing the specific number of contigs, as indicated by the x-axis. C-D) Sample plots as panels C and D, but for the whole metagenomic sample set.*\label{clustercontigqc}](../figures/contiglengthbycluster.png){ width=75% }

\newpage

![*Diversity calculations comparing cancer states of the colorectal virome, based on relative abundance of operational genomic units in each sample. A) NMDS ordination of community samples, colored for cancerous (green), pre-cancerous (red), and healthy (yellow). B) Differences in means between disease group centroids with 95% confidence intervals based on an ANOSIM test with a post hoc multivariate Tukey test. Comparisons (indicated on y-axis) in which the intervals cross the zero mean difference line (dashed line) were not significantly different. C) Shannon diversity and D) richness alpha diversity quantification comparing pre-cancerous (grey), cancerous (red), and healthy (tan) states.*\label{betaogu}](../figures/diversity-beta-ogu.pdf){ width=100% }

\newpage

![*Beta-diversity comparing disease states and the study negative controls. A) NMDS ordination of community samples, colored by disease state. B) Differences in means between disease group centroids with 95% confidence intervals based on an ANOSIM test with a post hoc multivariate Tukey test. Comparisons in which the intervals cross the zero mean difference line (dashed line) were not significantly different.*\label{betaogunegative}](../figures/diversity-beta-ogu-negative.pdf){ width=100% }

\newpage

![*Comparison of bacterial 16S rRNA classification models with and without OTUs whose median relative abundance are greater than zero. A) Classification model performance (measured as area under the curve) for bacteria models using 16S rRNA data both with and without filtering of samples whose median was zero. Significance was calculated using a Wilcoxon rank sum test, and the resulting p-value is shown. The random area under the curve (0.5) is marked with a dashed line. B) Relative abundance of the six bacterial OTUs removed when filtered for OTUs with median relative abundance of zero. OTU relative abundance is seperated by healthy (red) and cancerous (grey) samples. Relative abundance of 1% is marked by the dashed line.*\label{16scompare}](../figures/filtered16S.pdf){ width=100% }

![*Transition of colorectal cancer importance through disease progression. A) Virus and B) 16S rRNA gene model performance (AUC) when discriminating all binary combinations of disease types. Blue line represents mean performance from multiple random iterations. C-E) Top ten important phage OGUs when classifying each combination of disease state, as measured by the mean decrease in accuracy metric. Mean is represented by a point, and bars represent standard error. Disease comparison is specified in the top left corner of each panel. F-H) Top ten important bacterial 16S rRNA gene OTUs for classifying each disease state combination.*\label{transitionmodels}](../figures/temporalmodels.pdf){ width=75% }

\newpage

![*ROC curves from A) virome and B) bacterial 16S three-class random forest models tuned on mean AUC. Each curve represents the ability of the specified class to be classified against the other two classes. C) Quantification of the mean AUC variation for each model based on 10 model iterations. A pairwise Wilcoxon test with a Bonferroni multiple hypothesis correction demonstrated that the models are significantly different (alpha = 0.01). D) Mean decrease in accuracy when virome operational genomic units and E) bacterial 16S OTUs are removed from the respective three-class classification models. Results based on 25 iterations. F) Relative abundance of the six most important virome OGUs in the model, with the most important on the right. Line indicates abundance mean.*\label{threewaymodel}](../figures/predmodel-threewayclassification.pdf){ width=80% }

\newpage

![*Lysogenic phage relative abundance in disease states. Phage OGUs were predicted to be either lytic or lysogenic, and the relative abundance of lysogenic phages was quantified and represented as a boxplot. No disease groups were statistically significant.*\label{replicationstyles}](../figures/lysogenic-relabund.pdf){ width=50% }

\newpage

# References
